File: common/src/main/java/com/datastrato/gravitino/dto/requests/MetalakeUpdatesRequest.java
/*
 * Copyright 2023 Datastrato.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.dto.requests;

import com.datastrato.gravitino.rest.RESTRequest;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

/** Represents a request containing multiple Metalake updates. */
@Getter
@EqualsAndHashCode
@ToString
public class MetalakeUpdatesRequest implements RESTRequest {

  @JsonProperty("updates")
  private final List<MetalakeUpdateRequest> updates;

  /**
   * Constructor for MetalakeUpdatesRequest.
   *
   * @param updates The list of Metalake update requests.
   */
  public MetalakeUpdatesRequest(List<MetalakeUpdateRequest> updates) {
    this.updates = updates;
  }

  /** Default constructor for MetalakeUpdatesRequest. */
  public MetalakeUpdatesRequest() {
    this(null);
  }

  /**
   * Validates each request in the list.
   *
   * @throws IllegalArgumentException if validation of any request fails.
   */
  @Override
  public void validate() throws IllegalArgumentException {
    updates.forEach(RESTRequest::validate);
  }
}

File: clients/client-python/gravitino/dto/requests/metalake_updates_request.py
"""
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
"""

from dataclasses import dataclass, field
from typing import List

from dataclasses_json import config

from gravitino.dto.requests.metalake_update_request import MetalakeUpdateRequest
from gravitino.rest.rest_message import RESTRequest


@dataclass
class MetalakeUpdatesRequest(RESTRequest):
    """Represents a request containing multiple Metalake updates."""

    _updates: List[MetalakeUpdateRequest] = field(metadata=config(field_name="updates"))

    def __init__(self, updates: List[MetalakeUpdateRequest]):
        """Constructor for MetalakeUpdatesRequest.

        Args:
            updates: The list of Metalake update requests.
        """
        self._updates = updates

    def validate(self):
        """Validates each request in the list.

        Raises:
            IllegalArgumentException if validation of any request fails.
        """
        for update in self._updates:
            update.validate()
